pageserver/src/walredo.rs

```diff
@@ -116,13 +114,7 @@ struct ProcessOutput {
 pub struct PostgresRedoManager {
     tenant_id: TenantId,
     conf: &'static PageServerConf,
-    /// Counter to separate same sized walredo inputs failing at the same millisecond.
-    #[cfg(feature = "testing")]
-    dump_sequence: AtomicUsize,
-
-    stdout: Mutex<Option<ProcessOutput>>,
-    stdin: Mutex<Option<ProcessInput>>,
-    stderr: Mutex<Option<ChildStderr>>,
+    redo_process: Mutex<Option<Arc<WalRedoProcess>>>,
```

Comment:
Maybe RwLock, to allow a bit more concurrency. Not sure if it matters.

Reply:
So, you're proposing tack RwLock first, then if we observe `None` ~~upgrade~~ give back read lock and acquire write lock?

Reply:
yeah

Reply:
https://github.com/neondatabase/neon/pull/5557/commits/08dc86ceee2a8ebcd310d6abb37f2744980bc271